docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md

```diff
@@ -0,0 +1,43 @@
+# This document lists known breaking changes in Roslyn after .NET 8 all the way to .NET 9.
+
+## `Span` and `ReadOnlySpan` overloads are applicable in more scenarios in C# 14 and newer
+
+***Introduced in Visual Studio 2022 version 17.13***
+
+C# 14 introduces new [built-in span conversions and type inference rules](https://github.com/dotnet/csharplang/issues/7905).
+This means that different overloads might be chosen compared to C# 13, and sometimes an ambiguity compile-time error
+might be raised because a new overload is applicable but there is no single best overload.
+
+The following example shows some ambiguities and possible workarounds.
+Note that another workaround is for API authors to use `OverloadResolutionPriorityAttribute`.
```

Comment:
>. [](http://example.com/codeflow?start=92&length=1)

"or compile with `-langversion:13` or earlier"?

Reply:
Hm, that's true for the whole section, not just these ambiguity examples. It should be implied by "C# 14 introduces...". It feels weird to suggest this workaround explicitly, we want people to upgrade langversions, not stay on old versions.